cmd/juju/controller/listmodels_test.go

```diff
@@ -265,6 +265,23 @@ func (s *BaseModelsSuite) TestModelsOwner(c *gc.C) {
 	s.checkAPICalls(c, "BestAPIVersion", "ListModels", "ModelInfo", "Close")
 }
 
+// TestModelsForAdmin tests that a model admin user will get model credential.
+// Credential will only appear in non-tabular format - either yaml or json.
+func (s *BaseModelsSuite) TestModelsWithCredentials(c *gc.C) {
+	for i, infoResult := range s.api.infos {
+		// let's say only some models will have credentials returned from api...
+		if i%2 == 0 {
+			infoResult.Result.CloudCredentialTag = "cloudcred-some-cloud_some-owner_some-credential"
+		}
+	}
```

Comment:
Are the names of our credentials also redundant in specifying the owner and the cloud? In which case 'aws' actually occurs at least 3x in the 'juju list-models  --format=yaml' output?

Not saying its bad, but I do wonder.

Do you have an example of running this command against a live controller so we can see what it looks like?

Reply:
As per above, I'd leave it as is.

FWIW, a known issue.. For example, see https://bugs.launchpad.net/juju/+bug/1567712

Reply:
Oops, was totally meant to paste (/me rolls eyes at self)... from live testing:

$ juju models --format=yaml
models:
- name: admin/controller
  short-name: controller
  model-uuid: 1f163a56-882f-415d-8403-90c27bb4a48d
  controller-uuid: 3657b5df-5cac-4265-84c8-90026ad2fa9a
  controller-name: mycontroller
  owner: admin
  cloud: aws
  region: us-east-1
  credential:
    name: my-aws-credentials
    owner: admin
    cloud: aws
  type: ec2
  life: alive
  status:
    current: available
    since: 25 seconds ago
  access: admin
  last-connection: just now
  sla-owner: admin
  agent-version: 2.4-beta1.1
- name: admin/default
  short-name: default
  model-uuid: f9d101ed-162e-403c-86f3-bc9e7a8406d0
  controller-uuid: 3657b5df-5cac-4265-84c8-90026ad2fa9a
  controller-name: mycontroller
  owner: admin
  cloud: aws
  region: us-east-1
  credential:
    name: my-aws-credentials
    owner: admin
    cloud: aws
  type: ec2
  life: alive
  status:
    current: available
    since: 24 seconds ago
  access: admin
  last-connection: 7 seconds ago
  sla-owner: admin
  agent-version: 2.4-beta1.1
current-model: default

Reply:
All formatting got lost with paste... Here is how our formatted output actually looks: https://pastebin.ubuntu.com/p/H5QV5nD6NR/